### PR TITLE
spd api support custom indicator

### PR DESCRIPTION
**Pull request by luomingmeng**
#### What type of PR is this?

Enhancements
#### What this PR does / why we need it:
extend spd api to support custom indicator beyond business and system scopes

### CODE REVIEW

File: pkg/apis/workload/v1alpha1/types.go [hunk 1]
```diff
@@ -59,15 +60,39 @@ type ServiceProfileDescriptorSpec struct {
 	// +optional
 	BaselinePercent *int32 `json:"baselinePercent,omitempty"`
 
+	// if multiple CustomIndicator are defined, it means that we should
+	// try to satisfy all of those indicator targets
+	// +optional
+	// +patchMergeKey=name
```
Comment (cheney-lin) [inline pkg/apis/workload/v1alpha1/types.go:66]: Is patchMergeKey take effected for CRDs?

File: pkg/apis/workload/v1alpha1/types.go [hunk 2]
```diff
@@ -119,12 +144,12 @@ type ServiceBusinessIndicatorSpec struct {
 	Indicators []Indicator `json:"indicators,omitempty"`
 }
 
-type TargetIndicatorName string
```
Comment (csfldf) [inline pkg/apis/workload/v1alpha1/types.go:122]: changing type name, will it cause modifications in codes using this type?

Comment (luomingmeng) [reply pkg/apis/workload/v1alpha1/types.go:122]: yes, i will change the code in other mr

File: pkg/apis/workload/v1alpha1/types.go [hunk 3]
```diff
@@ -59,17 +60,41 @@ type ServiceProfileDescriptorSpec struct {
 	// +optional
 	BaselinePercent *int32 `json:"baselinePercent,omitempty"`
 
+	// if multiple CustomIndicator are defined, it means that we should
+	// try to satisfy all of those indicator targets
+	// +optional
+	// +listMapKey=name
+	// +listType=map
+	CustomIndicator []ServiceCustomIndicatorSpec `json:"customIndicator,omitempty"`
```
Comment (waynepeking348) [inline pkg/apis/workload/v1alpha1/types.go:68]: what is the difference between `CustomIndicator` and `BusinessIndicator` actually?

**Review by waynepeking348 — APPROVED** [review]

**Review by csfldf — APPROVED** [review]

**Pull request merged**